### ISSUE TITLE
Duplicate Project - project index settings are not duplicated correctly when an setting fail to get the configured value

### ISSUE THREAD

**Issue by yacdaniel** [craftercms/craftercms#6583]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When get settings fail to get the configured value of a project index setting, it would use the defaults settings instead. 

### Steps to reproduce

Steps:

1. Create a Project `ed` from any blueprint 
2. Modify Authoring's index settings 
```
curl -XPUT "http://localhost:9201/ed-authoring_v1/_settings" -H 'Content-Type: application/json' -d'
{
  "index.plugins.index_state_management.rollover_skip": true,
  "index": {
    "number_of_replicas": 4
  }
}'
```

3. Verify the index settings are updated `curl -s http://localhost:9201/ed-authoring_v1/_settings`
4. Duplicate the Project `ed` to `dup-ed`
5. Verify the index settings in the new duplicated Project 
```
curl -XPUT "http://localhost:9201/dup-ed-authoring_v1/_settings" -H 'Content-Type: application/json' -d'
{
  "index.plugins.index_state_management.rollover_skip": true,
  "index": {
    "number_of_replicas": 4
  }
}'
```
6. Notice that the index settings in the duplicated project are not equal to the original project `ed` settings  `curl -s http://localhost:9201/dup-ed-authoring_v1/_settings`

**Comment by yacdaniel** [comment]
Verified fixed in `4.1.3-SNAPSHOT`

**Comment by yacdaniel** [comment]
Verified fixed in `4.2.0-SNAPSHOT`
